### ISSUE TITLE
Problem with signing using Ledger account on Ethereum after trying to sign on unsupported network

### ISSUE THREAD

**Issue by michalinacienciala** [tahowallet/extension#3681]
I have a problem with signing using Ledger account on Ethereum after trying to sign on unsupported network (low priority, as it can be fixed by disconnecting dapp from the extension).

My steps:

1. Uninstall Taho
2. Go to https://link3.to/, disconnect (if connected before) and do a hard reset with cleaning cache. Close
3. Install Taho and import a Ledger account
4. Turn ON the testnets and switch to Arb Sepolia
5. Set Taho as a default wallet
6. Go to https://link3.to/ and choose `Connect wallet`. Approve the connection in wallet.
7. Choose `Send message` -> `Sign`

The wallet does not connect, There are these errors in the wallet logs:
```
chrome-extension://eajafomhmkipbjmfmhebemolkcicgfmd/background.js:118544:24 

Logs (wallet):
Async thunk failed Error: Unsupported network for Ledger signing
    at LedgerService.signMessage (index.ts:550:13)
    at SigningService.signData (index.ts:311:49)
    at main.ts:982:54
    at index.js:306:13
    at Array.map (<anonymous>)
    at Emittery.emit (index.js:304:23)
    at async signing.ts:86:40
    at async utils.ts:168:9

 ProviderBridgeService.routeContentScriptRPCRequest
```
This I think is expected, because Arbitrum Sepolia is not on a list of networks supported when signing on Ledger (https://github.com/tahowallet/extension/blob/main/background/constants/networks.ts#L178)

But then when I switch to Ethereum and reload the dapp, I'm still having the same issue. Only disconnecting the dapp from the wallet fixes that. **I'm not sure if that's a bug or is it how it's supposed to work.**
My steps (picking up after step 7):
  
8. Switch to Ethereum
9. Refresh link3.to
10. Choose `Connect wallet`.
11. Send message -> Sign. **After that the same errors as after step 7 are shown.**
12. Disconnect link3.to in the extension
13. Reload link3.to
14. Connect wallet.
15. Send message -> Sign.
16. Approve the signing in Ledger. This finally results in successful connection.
 
Tested using Taho 0.51.02922c34, Chrome & Brave
